classes/issue/Collector.php

```diff
@@ -328,14 +326,7 @@ public function getQueryBuilder(): Builder
         );
 
         // Context
-        // Never permit a query without a context_id unless the PKPApplication::CONTEXT_ID_ALL wildcard
-        // has been set explicitly.
-        if (!isset($this->contextIds)) {
-            throw new Exception('Submissions can not be retrieved without a context id. Pass the Application::CONTEXT_ID_ALL wildcard to get submissions from any context.');
```

Comment:
This check was coded to make it intentionally harder to forget the context ID filter; I think it's a good idea. It could be coded a different way (e.g. requiring `[null]` to be set as a context ID filter), but I'd rather it wasn't totally removed.

Reply:
I removed because I see this as a kind of exception treatment. Even though the context represents a big wall, data might be leaked in many different ways/places, and other places might be missing such checks.
But nothing against reverting the changes, just wanted to raise this small discussion.

Reply:
It's true that we're not applying these checks universally (and it would be tough to set a firm policy on when to apply them) -- but they are present here for issues in OJS and [submissions in pkp-lib](https://github.com/pkp/pkp-lib/blob/main/classes/submission/Collector.php#L364). Those are arguably the two most critical access points for published data (and by extension, related entities like publications, files, etc). They're just a few lines of code, and I don't see much of a downside. I'd feel better if we didn't remove them.

Reply:
Ok! In general, I'm against such protections, I'd like to think programmers know what they're doing, but we fail more often that we'd like to admit 😂